static/app/views/performance/landing/widgets/transforms/transformMetricsToArea.tsx

```diff
@@ -69,59 +69,94 @@ export function transformMetricsToArea<T extends WidgetDataConstraint>(
       )
     : undefined;
 
-  const data = groups.map(group => {
-    const series = response.intervals.map((intervalValue, intervalIndex) => {
-      const serieBucket = group.series[metricsField][intervalIndex];
-      const totalSerieBucket = totalPerBucket?.[intervalIndex];
-      return {
-        name: moment(intervalValue).valueOf(),
-        value:
-          defined(totalSerieBucket) &&
-          defined(serieBucket) &&
-          serieBucket > 0 &&
-          totalSerieBucket > 0
-            ? serieBucket / totalSerieBucket
-            : serieBucket,
-      };
-    });
+  const totalFailurePerBucket = isFailureRateWidget
+    ? response.intervals.map((_intervalValue, intervalIndex) =>
+        groups.reduce(
+          (acc, group) => acc + (group.series[metricsField][intervalIndex] ?? 0),
+          0
+        )
+      )
+    : undefined;
 
-    return {
-      seriesName: metricsField,
-      totals: group.totals[metricsField],
-      data: series.some(serie => defined(serie.value)) ? series : [],
-    };
-  });
+  const data = (
+    isFailureRateWidget
+      ? [
+          {
+            seriesName: 'failure_rate()',
+            data: response.intervals.map((intervalValue, intervalIndex) => {
+              const totalSerieBucket = totalPerBucket?.[intervalIndex];
+              const totalFailureSerieBucket = totalFailurePerBucket?.[intervalIndex];
+
+              return {
+                name: moment(intervalValue).valueOf(),
+                value:
+                  defined(totalFailureSerieBucket) &&
+                  defined(totalSerieBucket) &&
+                  totalSerieBucket > 0 &&
+                  totalFailureSerieBucket > 0
```

Comment:
We need to remove this line, the failure count can be 0.

Reply:
I have these conditions to avoid `NaN` and `Infinity`

Reply:
We can still check if totalFailureSerieBucket is a number, but it does not need to be greater than 0.
Consider this case: We have 20 total transactions, 0 failed. We want the failure rate to be 0/20 -> 0, not 20.

Reply:
but we can monitor it

Reply:
and if both are 0? example `0/0`, then we would get `NaN`